mhctools/netmhcii_pan.py

```diff
@@ -91,3 +98,125 @@ def prepare_allele_name(self, allele_name):
                 beta.gene,
                 beta.allele_family,
                 beta.allele_code)
+
+
+# TODO: rename this class to NetMHCIIpan3, turn this into a wrapper that figures out which version
+# of NetMHCIIpan is installed and use the appropriate wrapper
+class NetMHCIIpan3(NetMHCIIpanBase):
+    def __init__(
+            self,
+            alleles,
+            default_peptide_lengths=[15, 16, 17, 18, 19, 20],
+            program_name="netMHCIIpan",
+            process_limit=-1,
+            extra_flags=[]):
+        NetMHCIIpanBase.__init__(
+            self,
+            alleles=alleles,
+            parse_output_fn=parse_netmhciipan_stdout,
+            default_peptide_lengths=default_peptide_lengths,
+            program_name=program_name,
+            process_limit=process_limit,
+            extra_flags=extra_flags)
+
+
+class NetMHCIIpan4(NetMHCIIpanBase):
+    def __init__(
+            self,
+            alleles,
+            default_peptide_lengths=[15, 16, 17, 18, 19, 20],
+            program_name="netMHCIIpan",
+            process_limit=-1,
+            mode="elution_score",
+            extra_flags=[]):
+        """
+        Wrapper for NetMHCIIpan 4.0, using a different parser.
+        """
+
+        if mode not in ['binding_affinity', 'elution_score', 'all_output']:
+            raise ValueError("Unsupported mode", mode)
+
+        # Default to including binding affinity data (-BA flag), though the score and %rank will
+        # still be EL-based
+        NetMHCIIpanBase.__init__(
+            self,
+            alleles=alleles,
+            program_name=program_name,
+            process_limit=process_limit,
+            parse_output_fn=partial(parse_netmhciipan4_stdout, mode=mode),
+            default_peptide_lengths=default_peptide_lengths,
+            extra_flags=['-BA'] + extra_flags)
+
+        self.mode = mode
```

Comment:
I'm a bit confused - where is `self.mode` actually used?

Reply:
I copied from NetMHCpan4 on this, but it's not used anywhere I think (in either case). Mode is passed as an arg to the parser though.